ancestrymanager/ancestryutil.go

```diff
@@ -0,0 +1,96 @@
+package ancestrymanager
+
+import (
+	"fmt"
+	"strings"
+
+	resources "github.com/GoogleCloudPlatform/terraform-validator/converters/google/resources"
+)
+
+// AssetParent derives a resource's parent from its ancestors.
+func AssetParent(cai *resources.Asset, ancestors []string) (string, error) {
+	if err := validateAncestors(ancestors); err != nil {
```

Comment:
If I understand correctly, the validation here is in case a user calls this with invalid inputs. What would you think about making GetAncestors also return a value for the parent?

I think this is probably fine as-is but it feels like we end up having to do additional input validation because of having it in a separate function.

Reply:
Alternately, this function could call GetAncestors internally rather than taking a list of ancestors - but I think that may feel more complicated.

Reply:
It would require to change the ancestry manager interface in that case. 
Changing the interface: GetAncestors ->   func Ancestors(...) (ancestors []string, parent string, error)